native/core/Cargo.toml

```diff
@@ -78,6 +78,7 @@ object_store = { workspace = true }
 url = { workspace = true }
 chrono = { workspace = true }
 parking_lot = "0.12.3"
+datafusion-objectstore-hdfs = { path = "../hdfs", optional = true}
```

Comment:
So the `core` needs `hdfs`? then should we place it under `native/core/hdfs` rather than `native/hdfs`?

Reply:
I dont have a strong opinion here, if we feel the `hdfs` crate should be inside of a `core` we can do it for sure.

Reply:
maybe ok for now